src/graphql/entries.ts

```diff
@@ -77,18 +78,105 @@ export async function updateEntry({
   return copyOfDataFromPreviewApp;
 }
 
-function updateRichTextField({
+interface RichTextLink {
+  block: (Entity & CollectionItem)[];
+  inline: (Entity & CollectionItem)[];
+}
+
+function isEntityLinkEmpty(obj: RichTextLink) {
+  return Object.values(obj).every((arr) => arr.length === 0);
+}
+
+async function processNode(
+  node: any,
+  entries: RichTextLink,
+  assets: RichTextLink,
+  entityReferenceMap: EntityReferenceMap,
+  locale: string
+) {
+  // Check if the node is an embedded entity
+  if (node.nodeType.includes('embedded')) {
+    if (node.data && node.data.target && node.data.target.sys) {
+      const id = node.data.target?.sys.id || '';
+      const updatedReference = {
+        sys: { id: id, type: 'Link', linkType: node.data.target.sys.type },
+      };
+      let ref;
+
+      // Use the updateReferenceEntryField or updateReferenceAssetField function to resolve the entity reference
+      if (node.data.target.sys.linkType === 'Entry') {
+        ref = await updateReferenceEntryField(null, updatedReference, entityReferenceMap, locale);
+      } else if (node.data.target.sys.linkType === 'Asset') {
+        ref = await updateReferenceAssetField({
+          referenceFromPreviewApp: null,
+          updatedReference,
+          entityReferenceMap,
+          locale,
+        });
+      }
+
+      // Depending on the node type, assign the resolved reference to the appropriate array
+      switch (node.nodeType) {
+        case 'embedded-entry-block':
+          if (ref) entries.block.push(ref);
```

Comment:
we could check if there is a ref before going into the switch, and just push inside the switch/case